util/cbfstool/flashmap/fmap.c: Fix fmaptool endianness bugs on BE

Sakoram opened the pull request.

Commit:
util/cbfstool/flashmap/fmap.c: fix fmaptool endianness bugs on BE

Signed-off-by: Marek Kasiewicz <marek.kasiewicz@3mdeb.com>

krystian-hebel commented on util/cbfstool/flashmap/fmap.c:
```diff
@@ -48,7 +48,7 @@ static int is_valid_fmap(const struct fmap *fmap)
 		return 0;
 	/* a basic consistency check: flash should be larger than fmap */
 	if (fmap->size <
```
```suggestion
	if (le32toh(fmap->size) <
```

krystian-hebel commented on util/cbfstool/flashmap/fmap.c:
```diff
@@ -184,7 +184,7 @@ int fmap_print(const struct fmap *fmap)
 	kv_pair_print(kv);
```
Can't mark proper lines in the review, but just above this there are multi-byte fields printed, they should have their byte orders reversed (base, size and nareas).

krystian-hebel replied:
The same goes for lines 196-203 for offset, size and flags, and 206 where the flags are read again.

krystian-hebel replied:
And 361, 366, 376 for base, size, nareas (the proper way would be to convert the left side, with `fmap->`, I think).

krystian-hebel commented on util/cbfstool/flashmap/fmap.c:
```diff
@@ -319,7 +319,7 @@ const struct fmap_area *fmap_find_area(const struct fmap *fmap,
 		return NULL;
 
-	for (i = 0; i < fmap->nareas; i++) {
+	for (i = 0; i < le16toh((fmap)->nareas); i++) {
```
```suggestion
	for (i = 0; i < le16toh(fmap->nareas); i++) {
```

krystian-hebel commented on util/cbfstool/flashmap/fmap.c:
```diff
@@ -428,13 +428,13 @@ static int fmap_append_area_test(struct fmap **fmap)
 		goto fmap_append_area_test_exit;
```
Can't point to the proper lines again: 417-420 should use `htoleXX` on the right side of `=`, except for `name`

Sakoram replied:
I'm not sure if this is really necessary. Those values are only used in lines 441-444, If I use htoleXX here, I would have to use leXXtoh in 441-444. This seems pointless.

krystian-hebel replied:
It has to be swapped twice. If someone were to attach a debugger to inspect the memory, he/she should be able to see exactly the same format as if it were written to a file between these two points. Also, these tests may change at some point, someone may add a detailed dump or actually write this data to a file. It is easier to fix the code now than to debug it later.

Sakoram replied:
Ok. Corrected.

krystian-hebel commented on util/cbfstool/flashmap/fmap.c:
```diff
@@ -428,13 +428,13 @@ static int fmap_append_area_test(struct fmap **fmap)
+	(*fmap)->nareas = htole16(nareas_orig);
 
 	total_size = sizeof(**fmap) + sizeof(test_area);
 	if (fmap_append_area(fmap,
```
Offset, size and flags below should have `leXXtoh`

krystian-hebel replied:
Fixed

miczyg1 commented:
@krystian-hebel @Sakoram please clean up the PR so that we can merge it

Commit:
util/cbfstool/flashmap/fmap.c: Fix endianness in test functions

Signed-off-by: Marek Kasiewicz <marek.kasiewicz@3mdeb.com>

Sakoram commented:
@miczyg1 I fixed everything @krystian-hebel  commented about, except one function. I commented why here: https://github.com/3mdeb/coreboot/pull/21#discussion_r535945841

Sakoram commented:
@miczyg1  fixed everything @krystian-hebel  commented about, except one function. I commented why here: https://github.com/3mdeb/coreboot/pull/21#discussion_r535945841

krystian-hebel committed:
util/cbfstool/flashmap/fmap.c: add missing htole16()

This isn't necessary in this particular case, but it is safer to add this
in case someone copies the code to other tests.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
